src/lib/annotations/doc/DocDrawingThread.js

```diff
@@ -184,6 +188,53 @@ class DocDrawingThread extends DrawingThread {
     }
 
     /**
+     * Binds custom event listeners for the dialog.
+     *
+     * @protected
+     * @return {void}
+     */
+    bindCustomListenersOnDialog() {
+        if (!this.dialog) {
+            return;
+        }
+
+        this.dialog.addListener('annotationcreate', () => {
+            this.emit('annotationevent', {
+                type: 'softcommit'
+            });
+        });
+        this.dialog.addListener('annotationdelete', () => {
+            this.emit('annotationevent', {
+                type: 'dialogdelete'
+            });
+        });
+    }
+
+    /**
+     * Creates the document drawing annotation dialog for the thread.
+     *
+     * @protected
+     * @override
+     * @return {void}
+     */
+    createDialog() {
+        if (this.dialog) {
```

Comment:
Curious as to why we would need this?

Reply:
The dialog on a drawing in progress is different than the dialog on a saved drawing. Upon saving a drawing thread, we create a new dialog that is calculated as a saved drawing. (This adds the 'user drew' label and removes the save button.)